OpenRA.Mods.Common/Traits/Radar/ProvidesRadar.cs

```diff
@@ -28,7 +28,7 @@ static bool UpdateActive(Actor self)
 			// Check if powered
 			if (self.IsDisabled()) return false;
 
-			return self.World.ActorsWithTrait<JamsRadar>().All(a => a.Actor.Owner.Stances[self.Owner] == Stance.Ally
+			return self.World.ActorsWithTrait<JamsRadar>().All(a => !a.Actor.Owner.Stances[self.Owner].HasStance(a.Actor.Info.TraitInfo<JamsRadarInfo>().JamsRadarStances)
```

Comment:
You'll need to change `JamsRadar` to store a public reference to the `Info`, and then use `a.Trait.Info.JamsRadarStances` for that to work.